lib/chef/knife/azurerm_base.rb

```diff
@@ -41,13 +48,10 @@ def self.included(includer)
       end
 
       def service
+        details = check_authentication_method()
```

Comment:
Method name should be `authentication_details` instead of `check_authentication_method` since it's returning the details required for authentication